ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -503,10 +524,54 @@ private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, Ad
 
     byte[] content = contentJSON.toString().getBytes(StandardCharsets.UTF_8);
     ServerErrorCode error = ServerErrorCode.No_Error;
-
     return new AdminResponseWithContent(adminRequest.getCorrelationId(), adminRequest.getClientId(), error, content);
   }
 
+  /**
+   * Handles {@link BlobIndexAdminRequest}.
+   * @param requestStream the serialized bytes of the request.
+   * @param adminRequest the {@link AdminRequest} received.
+   * @return the {@link AdminResponseWithContent} to the request.
+   */
+  private AdminResponse handleBlobIndexRequest(DataInputStream requestStream, AdminRequest adminRequest) {
+    final int correlationId = adminRequest.getCorrelationId();
+    final String clientId = adminRequest.getClientId();
+    BlobIndexAdminRequest blobIndexAdminRequest = null;
+    String errorMessage = null;
+    try {
+      blobIndexAdminRequest = BlobIndexAdminRequest.readFrom(requestStream, adminRequest, storeKeyFactory);
+    } catch (Exception e) {
+      logger.error("Failed to deserialize BlobIndexAdminRequest", e);
+      return new AdminResponseWithContent(correlationId, clientId, ServerErrorCode.Bad_Request, null);
+    }
+
+    BlobId blobId = (BlobId) blobIndexAdminRequest.getStoreKey();
+    try {
+      PartitionId partitionId = blobId.getPartition();
+      Store store = storeManager.getStore(partitionId);
+      if (store == null) {
+        logger.error("BlobId {}'s partition[{}] doesn't exist or stopped in this host", blobId, partitionId.getId());
+        return new AdminResponseWithContent(correlationId, clientId, ServerErrorCode.Replica_Unavailable, null);
+      }
+
+      Map<String, MessageInfo> messages = store.findAllMessageInfosForKey(blobId);
+      // Serialize with JSON
+      String json = objectMapper.writeValueAsString(messages);
```

Comment:
Just for my understanding. Would this return empty string if input is empty map?

Reply:
it will return "{}" if the input the empty map.